sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/SyncBenchmark.java

```diff
@@ -0,0 +1,359 @@
+    private final ScheduledReporter reporter;
+    private final ExecutorService executorService;
+
+    private Meter successMeter;
+    private Meter failureMeter;
+
+    final Logger logger;
+    final CosmosClient cosmosClient;
+    final CosmosContainer cosmosContainer;
+
+    final String partitionKey;
+    final Configuration configuration;
+    final List<PojoizedJson> docsToRead;
+    final Semaphore concurrencyControlSemaphore;
+    Timer latency;
+
+    abstract class ResultHandler<T, Throwable> implements BiFunction<T, Throwable, T> {
+        ResultHandler() {
+        }
+
+        protected void init() {
+        }
+
+        @Override
+        abstract public T apply(T o, Throwable throwable);
+    }
+
+    class LatencyListener<T> extends ResultHandler<T, Throwable> {
+        private final ResultHandler<T, Throwable> baseFunction;
+        Timer.Context context;
+        LatencyListener(ResultHandler<T, Throwable> baseFunction, Timer latency) {
+            this.baseFunction = baseFunction;
+        }
+
+        protected void init() {
+            super.init();
+            context = latency.time();
+        }
+
+        @Override
+        public T apply(T o, Throwable throwable) {
+            context.stop();
+            return baseFunction.apply(o, throwable);
+        }
+    }
+
+    SyncBenchmark(Configuration cfg) throws Exception {
+        executorService = Executors.newFixedThreadPool(cfg.getConcurrency());
+
+        cosmosClient = new CosmosClientBuilder()
+            .setEndpoint(cfg.getServiceEndpoint())
+            .setKey(cfg.getMasterKey())
+            .setConnectionPolicy(cfg.getConnectionPolicy())
+            .setConsistencyLevel(cfg.getConsistencyLevel())
+            .buildClient();
+
+        cosmosContainer = cosmosClient.getDatabase(cfg.getDatabaseId()).getContainer(cfg.getCollectionId()).read().getContainer();
+
+        logger = LoggerFactory.getLogger(this.getClass());
+
+        partitionKey = cosmosContainer.read().getProperties().getPartitionKeyDefinition()
+            .getPaths().iterator().next().split("/")[1];
+
+        concurrencyControlSemaphore = new Semaphore(cfg.getConcurrency());
+        configuration = cfg;
+
+        ArrayList<CompletableFuture<PojoizedJson>> createDocumentFutureList = new ArrayList<>();
+
+        if (configuration.getOperationType() != Configuration.Operation.WriteLatency
+                && configuration.getOperationType() != Configuration.Operation.WriteThroughput
+                && configuration.getOperationType() != Configuration.Operation.ReadMyWrites) {
+            String dataFieldValue = RandomStringUtils.randomAlphabetic(cfg.getDocumentDataFieldSize());
+            for (int i = 0; i < cfg.getNumberOfPreCreatedDocuments(); i++) {
+                String uuid = UUID.randomUUID().toString();
+                PojoizedJson newDoc = generateDocument(uuid, dataFieldValue);
+                CompletableFuture futureResult = CompletableFuture.supplyAsync(() -> {
```

Comment:
Please specify parameters in this declaration as this is a parametrized class. 
Should be -> `CompletableFuture< PojoizedJson>`

Reply:
addressed